staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -276,6 +277,9 @@ func (o *LogsOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []str
 		}
 		infos, err := builder.Do().Infos()
 		if err != nil {
+			if apierrors.IsNotFound(err) {
+				err = fmt.Errorf("error from server (NotFound): %w in namespace %q", err, o.Namespace)
```

Comment:
```suggestion
				err = fmt.Errorf("error from server (NotFound): %v in namespace %s", err, o.Namespace)
```
seems better but output is equivalent anyway.